modules/libcom/src/misc/epicsString.c

```diff
@@ -358,3 +360,60 @@ unsigned int epicsMemHash(const char *str, size_t length, unsigned int seed)
     }
     return hash;
 }
+
+/* Compute normalized Levenshtein distance
+ *
+ * https://en.wikipedia.org/wiki/Levenshtein_distance
+ */
+double epicsStrSimilarCmp(const char *A, const char *B)
+{
+    double ret = 0;
+    size_t lA, lB, a, b;
+    size_t lmax;
+    size_t *dist0, *dist1, *stemp;
+
+    lA = strlen(A);
+    lB = strlen(B);
+    lmax = lA > lB ? lA : lB;
+
+    dist0 = calloc(1+lB, sizeof(*dist0));
+    dist1 = calloc(1+lB, sizeof(*dist1));
+    if(!dist0 || !dist1) {
+        ret = -1.0;
+        goto done;
+    }
+
+    for(b=0; b<1+lB; b++)
+        dist0[b] = b;
+
+    for(a=0; a<lA; a++) {
+        dist1[0] = a+1;
+
+        for(b=0; b<lB; b++) {
+            size_t delcost = dist0[b+1] + 1,
+                   inscost = dist1[b] + 1,
+                   subcost = dist0[b],
+                   mincost = delcost;
+
+            if(A[a]!=B[b])
+                subcost++;
```

Comment:
hum.  I hadn't considered this.  Maybe multiply the current (integer) costs by 2x so the case in-sensitive comparison can be added in as (effectively) one half.  Since the cost is propagated without being tested, and the output is normalized, I think this could work.

I'll think about it.  If this turns out to be difficult, I'm inclined to leave it as a future improvement.

Reply:
Given that `lmax` which is used for normalization is initially set to the larger string length I concluded there might be other places where a factor might be needed. Also I wasn't sure if 2x was the right factor, probably best make that a macro and see what effect it has. Thanks for considering this.

Reply:
Giving half weight to case insensitive comparison seems to work without much effort.  I've updated the examples in https://github.com/epics-base/epics-base/pull/104#issue-567842371